File: android/app/src/main/java/com/meta/wearable/dat/externalsampleapps/cameraaccess/data/ExerciseSet.kt
package com.meta.wearable.dat.externalsampleapps.cameraaccess.data

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "exercise_sets",
    foreignKeys = [
        ForeignKey(
            entity = WorkoutSession::class,
            parentColumns = ["id"],
            childColumns = ["sessionId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index("sessionId")]
)
data class ExerciseSet(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val sessionId: Long,
    val exerciseName: String = "",
    val repCount: Int = 0,
    val startTime: Long = System.currentTimeMillis(),
    val endTime: Long? = null,
    val guideImagePath: String? = null
)

File: ios/GymBro/Data/Models/ExerciseSet.swift
import Foundation
import SwiftData

@Model
final class ExerciseSet {
    var exerciseName: String
    var repCount: Int
    var startTime: Date
    var endTime: Date?
    var guideImagePath: String?
    var session: WorkoutSession?

    init(
        exerciseName: String = "",
        repCount: Int = 0,
        startTime: Date = Date(),
        endTime: Date? = nil,
        guideImagePath: String? = nil,
        session: WorkoutSession? = nil
    ) {
        self.exerciseName = exerciseName
        self.repCount = repCount
        self.startTime = startTime
        self.endTime = endTime
        self.guideImagePath = guideImagePath
        self.session = session
    }
}
